src/Test/Utilities/DisposableFile.cs

```diff
@@ -51,5 +53,31 @@ public void Dispose()
                 }
             }
         }
+
+        [DllImport("kernel32.dll", PreserveSig = false)]
```

Comment:
@davkean this code is only used in our unit tests hence it doesn't need to be fully portable at the moment.  It will definitely need to be revisited when we move our unit tests to core-clr though.

This code originated from [FileUtilities.cs](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Desktop/FileUtilities.cs#L360).  I noticed that it had all of the portability problems you mentioned but after discovering it was test only code I just moved it over here.